core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4QueryHttpSigner.java

```diff
@@ -31,11 +55,135 @@ public final class DefaultAwsV4QueryHttpSigner implements AwsV4QueryHttpSigner {
 
     @Override
     public SyncSignedRequest sign(SyncSignRequest<? extends AwsCredentialsIdentity> request) {
-        throw new UnsupportedOperationException();
+        // anonymous credentials, don't sign
+        if (CredentialUtils.isAnonymous(request.identity())) {
+            return SyncSignedRequest.builder()
+                .request(request.request())
+                .payload(request.payload().orElse(null))
+                .build();
+        }
+
+        String contentHash = calculateContentHash(request.payload().orElse(null), null);
+
+        SdkHttpRequest signedRequest = doSign(request, contentHash).build();
+        return SyncSignedRequest.builder()
+            .request(signedRequest)
+            .payload(request.payload().orElse(null))
+            .build();
+    }
+
+    private SdkHttpRequest.Builder doSign(SignRequest<?, ? extends AwsCredentialsIdentity> request, String contentHash) {
+        SdkHttpRequest.Builder requestBuilder = request.request().toBuilder();
+
+        Instant requestSigningInstant = validatedProperty(request, REQUEST_SIGNING_INSTANT);
+        Instant expirationInstant = validatedProperty(request, EXPIRATION_INSTANT,
+            requestSigningInstant.plusSeconds(SignerConstant.PRESIGN_URL_MAX_EXPIRATION_SECONDS));
+        Boolean doubleUrlEncode = validatedProperty(request, DOUBLE_URL_ENCODE, true);
+        Boolean normalizePath = validatedProperty(request, NORMALIZE_PATH, true);
+        String formattedRequestSigningDate = formatDateStamp(requestSigningInstant);
+        String formattedRequestSigningDateTime = formatTimestamp(requestSigningInstant);
+        String regionName = validatedProperty(request, REGION_NAME);
+        String serviceSigningName = validatedProperty(request, SERVICE_SIGNING_NAME);
+        String scope = buildScope(formattedRequestSigningDate, serviceSigningName, regionName);
+
+        long expirationInSeconds = getSignatureDurationInSeconds(requestSigningInstant, expirationInstant);
+
+        addHostHeader(requestBuilder);
+
+        AwsCredentialsIdentity sanitizedCredentials = sanitizeCredentials(request.identity());
+        if (sanitizedCredentials instanceof AwsSessionCredentialsIdentity) {
+            // For SigV4 query-signing, we need to add "X-Amz-Security-Token"
+            // as a query string parameter, before constructing the canonical
```

Comment:
hmm, the public docs https://docs.aws.amazon.com/IAM/latest/UserGuide/create-signed-request.html#temporary-security-credentials make it seem that this depends on service. So I was thinking this might become a property. But that section in fact seems to apply for when `Authorization` header is used too, i.e., AwsV4HttpSigner, but looks like even there we _always_ do it. So as long as it is carrying over current logic, seems ok.